### DISCUSSION TITLE
Prisma sortBy using extends computed Value

### DISCUSSION THREAD

**Discussion by Ejayz** [prisma/prisma#17990] [Q&A]
Hi i have this code here which takes 3 fields in database then join them: ```

>  export const prismaCustomTbl_Users = new PrismaClient({
>   log: ["query"],
> }).$extends({
>   result: {
>     tbl_users: {
>       name: {
>         needs: {
>           first_name: true,
>           middle_name: true,
>           last_name: true,
>         },
>         compute(tbl_users) {
>           return `${tbl_users.first_name} ${tbl_users.middle_name} ${tbl_users.last_name}`;
>         },
>       },
>     },
>   },
> });

 Now i want to sort data base on ``name`` but it gives error 
> `PrismaClientValidationError: 
> Invalid `prisma.tbl_users.findMany()` invocation:
> 
> {
>   where: {
>     NOT: {
>       user_id: 1
>     },
>     AND: {
>       is_exist: 'true'
>     }
>   },
>   orderBy: {
>     name: 'desc'
>     ~~~~
>   },
>   skip: 0,
>   take: 5,
>   select: {
>     user_id: true,
>     username: true,
>     name: true,
>     job: true,
>     phone: true
>   }
> }
> 
> Unknown arg `name` in orderBy.name for type tbl_usersOrderByWithRelationInput. Did you mean `username`? Available args:
> type tbl_usersOrderByWithRelationInput {
>   user_id?: SortOrder
>   username?: SortOrder
>   password?: SortOrder
>   first_name?: SortOrder
>   middle_name?: SortOrder
>   last_name?: SortOrder
>   phone?: SortOrder
>   job?: SortOrder
>   is_exist?: SortOrder
> }
> 
> 
>     at Yn.validate (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:148:91)
>     at Ur.createMessage (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:163:13577)
>     at C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:173:10192
>     at runInChildSpan (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:70:25749)
>     at t._executeRequest (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:173:10181)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async t._request (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\@prisma\client\runtime\library.js:173:9714)
>     at async GetUsers (webpack-internal:///(api)/./pages/api/get/UserManagement/[page].ts:48:18)
>     at async handler (webpack-internal:///(api)/./pages/api/get/UserManagement/[page].ts:28:22)
>     at async Object.apiResolver (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\api-utils\node.js:363:9)        
>     at async DevServer.runApi (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\next-server.js:487:9)
>     at async Object.fn (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\next-server.js:749:37)
>     at async Router.execute (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\router.js:253:36)
>     at async DevServer.run (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\base-server.js:384:29)
>     at async DevServer.run (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\dev\next-dev-server.js:741:20)       
>     at async DevServer.handleRequest (C:\Users\Ejayz-Super\Documents\Projects\webbased-piggery-management\node_modules\next\dist\server\base-server.js:322:20) {   
>   clientVersion: '4.10.1'
> }
> 
>

**Comment by nurul3101** [comment]
Hello @Ejayz 👋 

This would be the expected behaviour because `name` field doesn't exist in the database. `name` is a computed field which is computed once records are obtained from the database. Prisma Client computes results on access, not on retrieval.

So, `orderBy` name field would throw an error as the field doesn't exist on the database. For reference, here are the [considerations](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#considerations-for-fields) while using the `result` from Prisma Client Extensions.

**Reply by Ejayz** [reply]
Hi . so my use case was i should sort data by the computed name which consist of first_name middle_name and last_name

**Reply by nurul3101** [reply]
In that case, can you sort the data after fetching the records based on the where filter?

**Reply by Ejayz** [reply]
that is in the result ? havent tried that .

**Reply by nurul3101** [reply]
You should get the computed name field in the result that you can use to sort later.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
